### PR TITLE
9169 Add idb log error handling

### PR DESCRIPTION
**Pull request by fungairino**
## What does this PR do?

- Part of #9169

## Future Work

- We should use this error wrapping in the other idb calls for other databases as well.

## Checklist

_Leave all that are relevant and check off as completed_

- [ x] Added jest or playwright tests and/or storybook stories

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: src/telemetry/logging.ts [hunk 1]
```diff
@@ -186,12 +224,9 @@ export async function appendEntry(entry: LogEntry): Promise<void> {
...
   }
 
-  const db = await openLoggingDB();
-  try {
+  await withLoggingDB(async (db) => {
     await db.add(ENTRY_OBJECT_STORE, entry);
-  } finally {
-    db.close();
-  }
+  }, "appendEntry");
```
Comment (grahamlangford) [inline src/telemetry/logging.ts:229]:
```suggestion
  }, appendEntry.name);
```

Comment (fungairino) [reply src/telemetry/logging.ts]: Done!

File: src/telemetry/reportError.ts [hunk 2]
```diff
@@ -38,6 +38,39 @@ interface ErrorReportOptions {
...
+export const getReportErrorAdditionalContext = () => {
+  // In case of service worker.
+  const documentContext =
+    typeof window === "undefined"
+      ? { url: "", referrer: "" }
+      : {
+          // Record original current url and referrer here before it is lost in the service worker.
+          url: window.location.href,
+          referrer: document.referrer,
+        };
```
Comment (grahamlangford) [inline src/telemetry/reportError.ts:50]: Potentially leave `url` and `referrer` out and instead flag that it's the service worker?

Comment (grahamlangford) [reply src/telemetry/reportError.ts:50]: Thinking this would be confusing for some of the log consumers

Comment (fungairino) [reply src/telemetry/reportError.ts:50]: I'd lean towards keeping the same metadata property schema eeverywhere, and we can tell if it's in the service worker from `getContextName()` which will be "background"

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**